src/models/transfers.js

```diff
@@ -231,24 +233,26 @@ function * fulfillTransfer (transferId, fulfillment) {
 
     // We don't know if notary is executing or rejecting transfer
     const isValidExecution = Boolean(transfer.execution_condition &&
-      Condition.testFulfillment(transfer.execution_condition, fulfillment))
+      transfer.execution_condition.type === fulfillment.getData().condition_fulfillment.type &&
+      Condition.testFulfillment(transfer.execution_condition, fulfillment.getData().condition_fulfillment))
     const isValidCancellation = Boolean(transfer.cancellation_condition &&
-      Condition.testFulfillment(transfer.cancellation_condition, fulfillment))
+      transfer.cancellation_condition.type === fulfillment.getData().condition_fulfillment.type &&
+      Condition.testFulfillment(transfer.cancellation_condition, fulfillment.getData().condition_fulfillment))
 
     if (isValidCancellation === isValidExecution) {
       throw new UnmetConditionError('ConditionFulfillment failed')
     }
 
-    const canExecute = transfer.state === 'prepared'
-    const canCancel = transfer.state === 'proposed' || transfer.state === 'prepared'
+    const canExecute = transfer.state === 'prepared' || transfer.state === 'executed'
```

Comment:
Talking with @naoitoi, we feel that changing `PUT` to `POST` is a larger design decision that we should discuss as a team in more detail. Maybe for now I can revert so we throw an error when the user tries to fulfill a transfer multiple times? While this technically doesn't work for `PUT`, it solves the current problem, where we execute a transfer multiple times. That way we can discuss how we want the API to behave in more detail without holding up this PR.